### PR TITLE
Changes to not skip block payment

### PR DESCRIPTION
**Pull request by sgaragagghu**
Clean and simple
#283

### CODE REVIEW

**Comment by sgaragagghu** [comment]
There are issues about blocked connections of miners?

**Comment by xrdavies** [comment]
You see the code, it will block there.

**Comment by xrdavies** [comment]
I think maybe this issue is related to miners are disconnected by pool randomly. We will see.

File: client/mining_common.h [hunk 1]
```diff
@@ -3,6 +3,7 @@
 
 #include <pthread.h>
 #include "block.h"
+#include "pool.h"
```
Comment (jonano614) [inline client/mining_common.h]: Is this include necessary here?

File: client/mining_common.h [hunk 2]
```diff
@@ -13,8 +14,9 @@
 #include <poll.h>
 #endif
 
-#define DATA_SIZE          (sizeof(struct xdag_field) / sizeof(uint32_t))
-#define BLOCK_HEADER_WORD  0x3fca9e2bu
+#define DATA_SIZE		(sizeof(struct xdag_field) / sizeof(uint32_t))
+#define BLOCK_HEADER_WORD	0x3fca9e2bu
+#define POOL_TASK_ARRAY_SIZE	(CONFIRMATIONS_COUNT >> 1)
```
Comment (jonano614) [inline client/mining_common.h]: What is the purpose of the new constant?

Comment (jonano614) [reply client/mining_common.h]: I don't fully understand the purpose of these changes.

**Comment by jonano614** [comment]
I don't fully understand the purpose of these changes.

**Comment by sgaragagghu** [comment]
Those arrays are circular buffers, I made it bigger to record more old tasks.
Anyway an easiest way to handle this issue (even if it sound strange that this is the issue) is to add in create_block after add_blobk if(res&&miner) {add_queue} and pop the block(and other infos thst you need to remember) in block_thread.

**Comment by sgaragagghu** [comment]
I'll change it to use a queue, I think it's better

**Comment by sgaragagghu** [comment]
@jonano614 now it is even simpler, i'm testing it actually on testnet

**Comment by sgaragagghu** [comment]
it's working

**Comment by sgaragagghu** [comment]
It's interesting that client and server both get POLLHUP but a peer should send FIN afaik to get POLLHUP on the other side.

**Comment by sgaragagghu** [comment]
i'll reopen it if the issue still won't be fixed is i think

**Pull request closed without merging**